src/spyglass/lfp/v1/analytic_signal.py

```diff
@@ -0,0 +1,95 @@
+    """
+
+@schema
+class AnalyticSignalSelection(dj.Manual):
+    definition = """
+    # Table that combines the LFP data, and parameters for generating analytic signals
+    -> LFPBand
+    -> IntervalList
+    -> AnalyticSignalParameters
+    ---
+    """
+
+@schema
+class AnalyticSignal(dj.Computed):
+    definition = """
+    -> AnalyticSignalSelection
+    ---
+    -> AnalysisNwbfile # the name of the nwb file with the analytic sinal & time stamps data
+    analytic_signal_object_id: varchar(40)  # the NWB object ID for loading this object from the file
+    """
+    def make(self, key):
+        # First, get the filtered lfp data
+        FilteredBand = (LFPBand() & key).fetch_nwb()[0]['filtered_data']
+        
+        # Next, select data using defined valid interval
+        start = (IntervalList() & key).fetch1('valid_times')[0][0]
+        end = (IntervalList() & key).fetch1('valid_times')[0][-1]
+        TimeStamps_selected = FilteredBand.timestamps[np.logical_and(FilteredBand.timestamps >= start,FilteredBand.timestamps <= end)]
+        FilteredBand_selected = FilteredBand.data[np.logical_and(FilteredBand.timestamps >= start,FilteredBand.timestamps <= end)]
+        
+        # Get analytic signal with the specified method (we're only using Hilbert transform for now)
+        filtered_analytic = np.empty(shape=(FilteredBand_selected.shape[0],0),dtype=np.complex_)
+        for lfp_electrode_id in range(FilteredBand_selected.shape[1]):
+            filtered_lfp_one_channel = np.array(FilteredBand_selected)[:,lfp_electrode_id]
+            analytic_signal_one_channel = hilbert(filtered_lfp_one_channel, axis=0)
+            filtered_analytic = np.column_stack((filtered_analytic,analytic_signal_one_channel))
+        
+        # Combine times stamps and analytic signal results into a dataframe
+        timestamps_selected_df = pd.DataFrame({"time stamps":TimeStamps_selected})
+        analytic_signal_df = pd.DataFrame(filtered_analytic,columns=['electrode '+str(e) for e in FilteredBand.electrodes.data[:]])
+        analytic_signal_results = pd.concat((timestamps_selected_df,analytic_signal_df),axis=1)
+        
+        # Create an analysis nwb file to save the results (analytic signal and corresponding time stamps)
+        analytic_signal_file_name = AnalysisNwbfile().create(key["nwb_file_name"])
+        key['analysis_file_name'] = analytic_signal_file_name
+        
+        # get object id while generating the nwb file
+        analytic_signal_object_id = AnalysisNwbfile().add_nwb_object(
+            analysis_file_name=key['analysis_file_name'],
+            nwb_object=analytic_signal_results,
+        )
+        AnalysisNwbfile().add(key["nwb_file_name"], analytic_signal_file_name)
+        key['analytic_signal_object_id'] = analytic_signal_object_id
+        self.insert1(key)
+        print(
+            '\nPopulated analytic signal table for nwb_file_name='+key['nwb_file_name']+' between {0}ms and {1}ms'.format(start,end)
```

Comment:
This could also be an f-string:
```python
f"\nPopulated analytic signal table for nwb_file_name= {key['nwb_file_name']} between {start}ms and {end}ms"
```
Also, is the `\n` at the start of the print useful?